akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -50,6 +53,33 @@ private[akka] final case class ConnectionPoolSettingsImpl(
 }
 
 object ConnectionPoolSettingsImpl extends SettingsCompanion[ConnectionPoolSettingsImpl]("akka.http.host-connection-pool") {
+
+  private[akka] def hostRegex(pattern: String): Regex = {
+    val regexPattern = if (pattern.startsWith("regex:")) {
+      pattern.stripPrefix("regex:")
+    } else {
+      pattern.stripPrefix("glob:").map {
+        case '*'   ⇒ ".*"
+        case '?'   ⇒ "."
+        case '.'   ⇒ "\\."
+        case other ⇒ other.toString
+      }.mkString
+    }
+
+    // If the pattern starts with a wildcard, we want to match subdomains, as well as the raw domain
```

Comment:
Is that a common assumption? I see how it can be useful but e.g. wildcard TLS certificates are *not* matched that way.

Reply:
I guess I hadn't thought about it too much. It just seemed intuitive to me, but yeah, I dunno